### DISCUSSION TITLE
updating model from route model binding

### DISCUSSION THREAD

**Discussion by jackade2022** [livewire/livewire#5182] [1. Help]
Hi, I have a problem when I add one property or array to the model from route model binding, after updating any route, in livewire this change is removed, how to save this attribute or array. THX

**Comment by jackade2022** [comment]
for example:
`
public function mount(User $user){
  $user -> test = ['jack']
}
`
### this user and its array exist


after is updated some a route ex:
`
public function action(){
dd($user) ;
}
`
### this user exist but without its array test

**Comment by joshhanley** [comment]
@jackade2022 Livewire won't send any of the extra data to the front end, hence why after an another Livewire request has happened, the data is missing. This is a security measure of Livewire to ensure sensitive data that may be stored in your models isn't sent to the front end by mistake.

You will need to add the properties of your model, that you want to persist between requests, to your `$rules` array. See here for more details https://laravel-livewire.com/docs/2.x/properties#binding-models

Hope this helps!

**Reply by jackade2022** [reply]
THx, but when I toke my User form the binding in mount and after that I added several relations, for example, the address user or orders...., these relation remaining with a user after each request but when I manually added some array or property, these are removed! but I don't want they were removed.
As my request is very heavy so I would remain extra property with the user.

**Reply by joshhanley** [reply]
@jackade2022 yes, that's correct, Livewire can restore relations, but it doesn't know about each individual relation records. It only know which relations were loaded, and calls load and passes the relation names in `->load(['address', orders'])`. See here for details https://laravel.com/docs/9.x/eloquent-relationships#lazy-eager-loading

If you need individual properties data to persist between requests, you need to add them to your `$rules` array like I suggested in my previous comment 🙂

**Reply by jackade2022** [reply]
Thank you, so much
